File: aircon/src/main/java/com/ironsource/aura/aircon/logging/AndroidLogger.java
package com.ironsource.aura.aircon.logging;

import android.util.Log;

/**
 * Default logger for the SDK, logs to logcat with "AirCon" tag.
 */
public class AndroidLogger
		implements Logger {

	private static final String TAG = "AirCon";

	@Override
	public void v(final String msg) {
		Log.v(TAG, msg);
	}

	@Override
	public void d(final String msg) {
		Log.d(TAG, msg);
	}

	@Override
	public void w(final String msg) {
		Log.w(TAG, msg);
	}

	@Override
	public void i(final String msg) {
		Log.i(TAG, msg);
	}

	@Override
	public void e(final String msg) {
		Log.e(TAG, msg);
	}

	@Override
	public void logException(final Exception e) {
		e.printStackTrace();
	}
}

File: aircon/src/main/java/com/ironsource/aura/airconkt/logging/AndroidLogger.kt
package com.ironsource.aura.airconkt.logging

import android.util.Log

private const val TAG = "AirConKt"

/**
 * Default logger for the SDK, logs to logcat with "AirConKt" tag.
 */
class AndroidLogger : Logger {

    override fun v(msg: String) {
        Log.v(TAG, msg)
    }

    override fun d(msg: String) {
        Log.d(TAG, msg)
    }

    override fun w(msg: String) {
        Log.w(TAG, msg)
    }

    override fun i(msg: String) {
        Log.i(TAG, msg)
    }

    override fun e(msg: String,
                   e: Exception?) {
        Log.e(TAG, msg)
        e?.printStackTrace()
    }
}